test-app/src/main/java/org/readium/r2/testapp/reader/ReaderActivity.kt

```diff
@@ -139,6 +139,9 @@ open class ReaderActivity : AppCompatActivity() {
             is ReaderViewModel.Event.Failure -> {
                 Toast.makeText(this, event.error.getUserMessage(this), Toast.LENGTH_LONG).show()
             }
+            else -> {
+                Toast.makeText(this, getString(R.string.error), Toast.LENGTH_LONG).show()
+            }
```

Comment:
This will show the toast when the user start a search in a book. I think instead we should simply add branches for all the available events.

Currently the `SearchFragment` is handling the `StartNewSearch` event which is dangerous as if we revert to a single `Channel` for event handling, it can by default have only a single observer. Best practice would be to handle all the events in the `ReaderActivity`, and forward downward any event if needed.

For this PR, I suggest we just move the `StartNewSearch` event to the `ReaderViewModel.FeedbackEvent` enum and let the `EpubReaderFragment` take ownership of this event handling, as it has access to the `SearchFragment`.

Reply:
`EpubReaderFragment` doesn't have a reference to `SearchFragment`, all it does show the fragment via the fragment manager, so it's not quite as easy as letting EpubReaderFragment take ownership of event handling for this.  But yeah that's an overlook on my part putting a toast error message when it results in showing that when a new search begins.  I think the best thing to do is just add an empty else branch.

Reply:
Sounds good for now, we can always fix this in a dedicated refactoring.